src/promptflow-devkit/promptflow/_sdk/operations/_flow_operations.py

```diff
@@ -647,23 +646,51 @@ def get_git_base_dir():
                 .strip()
             )
 
-        dependencies = ["promptflow-devkit", "promptflow-core", "promptflow-tracing"]
-        # get promptflow-** required and extra packages
-        extra_packages = []
-        required_packages = []
-        for package in dependencies:
-            with open(get_git_base_dir() / "src" / package / "pyproject.toml", "rb") as file:
-                data = toml.load(file)
-            extras = data.get("tool", {}).get("poetry", {}).get("extras", {})
-            for _, package in extras.items():
-                extra_packages.extend(package)
-            requires = data.get("tool", {}).get("poetry", {}).get("dependencies", [])
-            for package, _ in requires.items():
-                required_packages.append(package)
+        def is_tool(name):
+            """Check whether `name` is on PATH and marked as executable."""
+
+            # from whichcraft import which
+            from shutil import which
+
+            return which(name) is not None
 
+        def get_package_dependencies(package_name_list):
+            dependencies = []
+            for package_name in package_name_list:
+                if is_tool("conda"):
+                    result = subprocess.run(
+                        "conda activate root | pip show {}".format(package_name), shell=True, stdout=subprocess.PIPE
+                    )
+                else:
+                    result = subprocess.run(["pip", "show", package_name], stdout=subprocess.PIPE)
```

Comment:
pip/conda is not guaranteed.

Reply:
offline synced, will use a separate txt file to hold packages for executable. And add a test to keep the file consistent with toml file of devkit/core/tracing